Crash due to nil *File while trying to lock Window

I can't reproduce this. The crash happened while middle clicking on `Del` on a window tag. The window was created by [acme-lsp](https://github.com/fhs/acme-lsp). Full stack trace: [edwood-crash.txt](https://github.com/rjkroege/edwood/files/3243301/edwood-crash.txt).
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xb8 pc=0x5acf62]

goroutine 34 [running, locked to thread]:
panic(0x60df20, 0x7e7af0)
        /home/big/go/src/runtime/panic.go:662 +0x2c2 fp=0xc0005a3d60 sp=0xc0005a3cd0 pc=0x42e862
runtime.panicmem(...)
        /home/big/go/src/runtime/panic.go:183
runtime.sigpanic()
        /home/big/go/src/runtime/signal_unix.go:391 +0x3ec fp=0xc0005a3d90 sp=0xc0005a3d60 pc=0x4436fc
main.(*File).AllText(0x0, 0xc0005a3dd0)
        /home/fhs/go/src/github.com/rjkroege/edwood/file.go:309 +0x22 fp=0xc0005a3dc0 sp=0xc0005a3d90 pc=0x5acf62
main.(*Window).Lock(0xc00088dc00, 0x4d)
        /home/fhs/go/src/github.com/rjkroege/edwood/wind.go:330 +0x8b fp=0xc0005a3df8 sp=0xc0005a3dc0 pc=0x5cafdb
main.MovedMouse(0x4d6, 0x22d, 0x0, 0x616e1e50)
        /home/fhs/go/src/github.com/rjkroege/edwood/acme.go:338 +0x71f fp=0xc0005a3e58 sp=0xc0005a3df8 pc=0x5932ef
main.mousethread(0x699fa0, 0xc0000101f8)
        /home/fhs/go/src/github.com/rjkroege/edwood/acme.go:300 +0x304 fp=0xc0005a3fd0 sp=0xc0005a3e58 pc=0x5928f4
runtime.goexit()
        /home/big/go/src/runtime/asm_amd64.s:1357 +0x1 fp=0xc0005a3fd8 sp=0xc0005a3fd0 pc=0x45bf01
created by main.main.func1
        /home/fhs/go/src/github.com/rjkroege/edwood/acme.go:170 +0x800
```

--- a/dat.go
+++ b/dat.go
@@ -89,9 +89,9 @@ var (
 
 	seltext   *Text
 	argtext   *Text
-	mousetext *Text
-	typetext  *Text
-	barttext  *Text
+	mousetext *Text // global because Text.Close needs to clear it
+	typetext  *Text // global because Text.Close needs to clear it
+	barttext  *Text // shared between mousethread and keyboardthread
 
 	bartflag          bool
 	swapscrollbuttons bool

--- a/xfid.go
+++ b/xfid.go
@@ -229,6 +229,10 @@ func xfidclose(x *Xfid) {
 	q = FILE(x.f.qid)
 	x.f.open = false
 	if w != nil {
+		// We need to lock row here before locking window (just like mousethread)
+		// in order to synchronize mousetext with mousethread: mousetext is
+		// set to nil when the associated window is closed.
+		row.lk.Lock()
 		w.Lock('E')
 		switch q {
 		case QWctl:
@@ -271,6 +275,7 @@ func xfidclose(x *Xfid) {
 		}
 		w.Close()
 		w.Unlock()
+		row.lk.Unlock()
 	} else {
 		switch q {
 		case Qeditout:

